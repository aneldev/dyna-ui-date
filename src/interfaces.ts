export type TContent = JSX.Element | string;

export enum EMonthCalendarColor {
  GREY_GREEN = "GREY_GREEN",
  GREY_CYAN = "GREY_CYAN",
}

