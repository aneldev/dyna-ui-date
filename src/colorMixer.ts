import {EColor as EPickerContainerColor, EColor as EPickerButtonColor, EColor as EFieldColor} from "dyna-ui-styles";
import {EMonthCalendarColor} from "./interfaces";

export interface IColorMixer {
  fieldColor?: EFieldColor;
  calendarColor?: EMonthCalendarColor,
  pickerButtonColor?: EPickerButtonColor,
  pickerContainerColor?: EPickerContainerColor,
}

export enum EColor {
  GREY_ORANGE_GREEN_CALENDAR_GREEN = "GREY_ORANGE_GREEN_CALENDAR_GREEN",
  GREY_RED_GREEN_CALENDAR_GREEN = "GREY_RED_GREEN_CALENDAR_GREEN",
  WHITE_BLACK_CALENDAR_GREEN = "WHITE_BLACK_CALENDAR_GREEN",

  GREY_ORANGE_GREEN_CALENDAR_CYAN = "GREY_ORANGE_GREEN_CALENDAR_CYAN",
  GREY_RED_GREEN_CALENDAR_CYAN = "GREY_RED_GREEN_CALENDAR_CYAN",
  WHITE_BLACK_CALENDAR_CYAN = "WHITE_BLACK_CALENDAR_CYAN",

  // this is the same with the xxx_CALENDAR_GREEN, is kept for backward compatibility
  GREY_ORANGE_GREEN = "GREY_ORANGE_GREEN",
  GREY_RED_GREEN = "GREY_RED_GREEN",
  WHITE_BLACK = "WHITE_BLACK",

}

export const colorMixer = (color: EColor): IColorMixer => {
  switch (color) {

    // green calendar
    case EColor.GREY_RED_GREEN_CALENDAR_GREEN:
    case EColor.GREY_RED_GREEN:
      return {
        calendarColor: EMonthCalendarColor.GREY_GREEN,
        fieldColor: EFieldColor.RED_WHITE,
        pickerButtonColor: EPickerButtonColor.RED_WHITE,
        pickerContainerColor: EPickerContainerColor.WHITE_RED,
      };
    case EColor.GREY_ORANGE_GREEN_CALENDAR_GREEN:
    case EColor.GREY_ORANGE_GREEN:
      return {
        calendarColor: EMonthCalendarColor.GREY_GREEN,
        fieldColor: EFieldColor.ORANGE_WHITE,
        pickerButtonColor: EPickerButtonColor.ORANGE_WHITE,
        pickerContainerColor: EPickerContainerColor.WHITE_ORANGE,
      };
    case EColor.WHITE_BLACK_CALENDAR_GREEN:
    case EColor.WHITE_BLACK:
      return {
        calendarColor: EMonthCalendarColor.GREY_GREEN,
        fieldColor: EFieldColor.BLACK_WHITE,
        pickerButtonColor: EPickerButtonColor.BLACK_WHITE,
        pickerContainerColor: EPickerContainerColor.WHITE_BLACK,
      };

    // clean calendar
    case EColor.GREY_RED_GREEN_CALENDAR_CYAN:
      return {
        calendarColor: EMonthCalendarColor.GREY_CYAN,
        fieldColor: EFieldColor.RED_WHITE,
        pickerButtonColor: EPickerButtonColor.RED_WHITE,
        pickerContainerColor: EPickerContainerColor.WHITE_RED,
      };
    case EColor.GREY_ORANGE_GREEN_CALENDAR_CYAN:
      return {
        calendarColor: EMonthCalendarColor.GREY_CYAN,
        fieldColor: EFieldColor.ORANGE_WHITE,
        pickerButtonColor: EPickerButtonColor.ORANGE_WHITE,
        pickerContainerColor: EPickerContainerColor.WHITE_ORANGE,
      };
    case EColor.WHITE_BLACK_CALENDAR_CYAN:
      return {
        calendarColor: EMonthCalendarColor.GREY_CYAN,
        fieldColor: EFieldColor.BLACK_WHITE,
        pickerButtonColor: EPickerButtonColor.BLACK_WHITE,
        pickerContainerColor: EPickerContainerColor.WHITE_BLACK,
      };
  }
};

