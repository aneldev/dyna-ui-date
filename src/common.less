.hover{
	transition: opacity ease-in-out 200ms;

	&:hover{
		opacity: 0.7;
	}
}

