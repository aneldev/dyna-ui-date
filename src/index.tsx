export {
  TContent,
}from "./interfaces";

export {
  DynaMonthCalendar,
  IDynaMonthCalendarProps,
} from "./DynaMonthCalendar/DynaMonthCalendar";

export {
  EMonthCalendarColor as EDynaMonthCalendarColor,
} from "./interfaces";

export {
  EMode,
  ESize, EStyle,
  DynaDatePicker,
  IDynaDatePickerProps,
} from "./DynaDatePickers/DynaDatePicker";

export {
  EEditDate,
  DynaDateRangePicker,
  IDynaDateRangePickerProps,
} from "./DynaDatePickers/DynaDateRangePicker";

export {
  EColor,
  EColor as EDynaDatePickerColor,
} from "./colorMixer";

