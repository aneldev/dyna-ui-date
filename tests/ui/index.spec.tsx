import "jest";

import { configure } from 'enzyme';
import * as Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });

import * as React from 'react';
import * as enzyme from 'enzyme';

import {DynaDatePicker} from '../../src';

describe('Home', () => {
  let wrapper;

  it('has expected content with deep render', () => {
    wrapper = enzyme.shallow(
      (
        <DynaDatePicker
          name="Departure date"
          onChange={()=>undefined}
        />
      ),
      {}
    );

    expect(wrapper).toMatchSnapshot();
  });
});
