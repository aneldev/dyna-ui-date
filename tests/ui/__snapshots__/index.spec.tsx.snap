// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home has expected content with deep render 1`] = `
<div
  className="dyna-date-picker dyna-date-picker-mode-EDIT dyna-date-picker-style-INLINE_ROUNDED dyna-date-picker-size-MEDIUM"
>
  <DynaFieldWrapper
    className={null}
    color="ORANGE_WHITE"
    footer={
      <DynaPickerContainer
        color="WHITE_ORANGE"
        pointerPosition="MIDDLE"
        responsive={true}
        show={false}
        style="INLINE_ROUNDED"
      >
        <div
          className="ddp-picker-container"
        >
          <div
            className="ddp--calendar--label"
          >
            <h2 />
          </div>
          <DynaMonthCalendar
            color="GREY_GREEN"
            end={null}
            hoverOn={null}
            hoverStart={null}
            max={null}
            min={null}
            mode="EDIT"
            name="Departure date"
            onChange={[Function]}
            onHover={[Function]}
            onViewportChange={[Function]}
            renderPickerDay={[Function]}
            renderPickerMonthYear={[Function]}
            renderPickerWeekDay={[Function]}
            staringFromWeekDay={1}
            start={null}
            value={null}
            values={Array []}
          />
          <div
            className="ddp--calendar--button-bar"
          >
            <div>
              <DynaButton
                className=""
                color="ORANGE_WHITE"
                disabled={false}
                ghost={false}
                href={null}
                onClick={[Function]}
                size="LARGE"
                style="ROUNDED"
                target="_blank"
              >
                <span>
                  <i
                    aria-hidden="true"
                    className="fa fa-calendar"
                  />
                   Today
                </span>
              </DynaButton>
            </div>
            <div>
              <DynaButton
                className=""
                color="ORANGE_WHITE"
                disabled={false}
                ghost={false}
                href={null}
                onClick={[Function]}
                size="LARGE"
                style="ROUNDED"
                target="_blank"
              >
                <span>
                  <i
                    aria-hidden="true"
                    className="fa fa-check"
                  />
                   Ok
                </span>
              </DynaButton>
            </div>
          </div>
        </div>
      </DynaPickerContainer>
    }
    inputElementSelector=".ddp-input-control"
    isLoading={null}
    label={null}
    mode="EDIT"
    onClick={[Function]}
    onFocus={[Function]}
    onOutsideClick={[Function]}
    required={null}
    size="MEDIUM"
    style="INLINE_ROUNDED"
    validationMessage={null}
  >
    <input
      className="ddp-input-control"
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      readOnly={true}
      value=""
    />
  </DynaFieldWrapper>
</div>
`;
